photon-api/src/main/scala/com/linkedin/photon/ml/supervised/model/GeneralizedLinearModel.scala

```diff
@@ -165,4 +159,35 @@ object GeneralizedLinearModel {
     broadcastModel.unpersist()
     result
   }
+
+  val MODEL_TYPE = "modelType"
+
+  /**
+   * A UDF to compute scores given a linear model and a feature vector
+   *
+   * @return The score which is the dot product of model coefficients and features
+   */
+  def scoreUdf = udf[Double, SparseVector, SparseVector](
+    { (coefficients: SparkVector, features: SparkVector) =>
+      require(
+        coefficients.size == features.size,
+        s"Coefficients.size = ${coefficients.size} and features.size = ${features.size}")
+
+      var score = 0D
+
+      coefficients match {
```

Comment:
If the features are dense, then the coefficients are usually dense. If the features are sparse (for random effect), then the coefficients are sparse. So it seems that
```
features.foreachActive { case (index, value) => score += value * denseCoef(index)}
```
is good enough. Will there be cases that coefficients are sparse but features are dense?